api/modules/ways/views.py

```diff
@@ -0,0 +1,282 @@
+            for j in range(0, all_types):
+                if address_components[i]['types'][j] == 'locality':
+                    city.append(address_components[i]['long_name'])
+
+        for i in range(0, all_address_components):
+            all_types = len(['types'])
+            for j in range(0, all_types):
+                if address_components[i]['types'][j] == 'administrative_area_level_1':
+                    city.append(address_components[i]['long_name'])
+        place_search_api_response = requests.get(
+            PLACE_SEARCH_API_URL.format(city, 'train_station')
+            )
+        place_search_api_response_json = place_search_api_response.json()
+
+        station_place_id = place_search_api_response_json['results'][0]['place_id']
+        station_name = place_search_api_response_json['results'][0]['name']
+        distance_api_response = requests.get(
+            DISTANCE_MATRIX_API_URL.format('place_id:'+place_id, 'place_id:'+station_place_id,  walk_or_drive='driving')
+            )
+        distance_api_response_json = distance_api_response.json()
+
+        distance = distance_api_response_json['rows'][0]['elements'][0]['distance']['text']
+        duration = distance_api_response_json['rows'][0]['elements'][0]['duration']['text']
+        response_string = '{} is {} away from {} station ' \
+                            'and can be reached in {} drive.'.format(destination, distance, station_name, duration)
+
+        if place_autocomplete_api_response.status_code == 503:
+            error_message = "Places Autocomplete API error - Service Unavailable"
+            return Response(error_message, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        if place_autocomplete_api_response.status_code == 401:
+            error_message = 'Places Autocomplete API error - Invalid authentication'
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+        if place_autocomplete_api_response.status_code == 403:
+            error_message = 'Places Autocomplete API error - Permission denied/Forbidden'
+
+            # create GitHub issue to get developers' attention
+            make_github_issue(error_message)
+
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+        if place_search_api_response.status_code == 503:
+            error_message = "Places Search API error - Service Unavailable"
+            return Response(error_message, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        if place_search_api_response.status_code == 401:
+            error_message = 'Places Search API error - Invalid authentication'
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+        if place_search_api_response.status_code == 403:
+            error_message = 'Places Search API error - Permission denied/Forbidden'
+
+            # create GitHub issue to get developers' attention
+            make_github_issue(error_message)
```

Comment:
This line is getting repeated several times. You can reduce it by using if else statement. In case there are any errors, make an issue.

Reply:
`make_github_issue(error_message)` is only called when api response status code is 403.
So do you want that whenever google maps api response is 403,  Flag is turned to true, and whenever the flag is turned true, 'make_github_issue(error_message)' be called? And hence 'make_github_issue(error_message)' will be called only once in whole route( one function) ?

Reply:
Yes. But the ```error_message``` is changing at many places so you can call ```make_github_issue(error_message)``` just once. You have called ```make_github_issue(error_message)``` several times which can be reduced to just one time.